src/PackageVersions/Installer.php

```diff
@@ -80,27 +80,33 @@ public static function getSubscribedEvents()
      */
     public static function dumpVersionsClass(Event $composerEvent)
     {
-        $io = $composerEvent->getIO();
+        $composer = $composerEvent->getComposer();
+        $versions = iterator_to_array(self::getVersions($composer->getLocker(), $composer->getPackage()));
 
-        $io->write('<info>ocramius/package-versions:</info>  Generating version class...');
+        if (!isset($versions['ocramius/package-versions'])) {
+            //plugin must be globally installed - we only want to generate versions for projects which specifically
```

Comment:
@Ocramius will run in both contexts as it would be in the lock file - is that expected?

The first is actually my use case - A globally installed tool depends on `PackageVersions`.